protocol/contracts/beanstalk/sun/SeasonFacet/Receiving.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {C} from "contracts/C.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {AppStorage, Storage} from "contracts/beanstalk/AppStorage.sol";
+import {LibFertilizer} from "contracts/libraries/LibFertilizer.sol";
+import {ReentrancyGuard} from "contracts/beanstalk/ReentrancyGuard.sol";
+
+/**
+ * @notice Constraints of how many Beans to send to a given route at the current time.
+ * @param points Weight of this shipment route relative to all routes.
+ * @param cap Maximum Beans that can be received by this stream at this time.
+ */
+struct ShipmentPlan {
+    uint256 points;
+    uint256 cap;
+}
+
+/**
+ * @title Receiving
+ * @author funderbrker
+ * @notice Holds the functions responsible for receiving Bean shipments after mints. These
+ *  functions must live inside of the Beanstalk Diamond. If new receiving components are needed,
+ *  this contract and its containing Facet will need to be updated.
+ * @dev An alternative design could remove the need for the generalized receive() entry function
+ *  and instead require the shipping route to define the selector of its own corresponding receive
+ *  function. However, both designs will require a Facet cut if a new receive function is needed,
+ *  so this design was chosen for additional clarity.
+ */
+contract Receiving is ReentrancyGuard {
```

Comment:
Since `Receiving` inherits `ReentrancyGuard`, `Oracle.sol` does not need to inherit `ReentrancyGuard` anymore. Could we remove that from oracle.sol?

Reply:
It is inheriting reentrancy guard because the AppStorage variable `s` is defined there. if `s` is also declared here it will conflict and break. i do not understand the design decision to place the declaration in reentrancy guard. could use some insight here

Reply:
`Oracle.sol` also uses this class variable

Reply:
Got it, lets leave for now and reevaluate later.